### ISSUE TITLE
[JENKINS-8423] Slave priority

### ISSUE THREAD

**Issue by jenkins-infra-bot** [timja-org/jenkins-gh-issues-poc-11-23#4214]
<p>Hello,</p>

<p>I have several slaves to execute my jobs.<br/>
Some are real user workstation, other are pure dedicated servers.</p>

<p>It would be great if I could say to Hudson:<br/>
"Use this dedicated slaves as much as you can. And if they are busy use the not dedicated ones."</p>

<p>With this I would be able to handle heavy loads (with the extra help of user workstations).<br/>
But when everything is calm, I won't bother the real users with my jobs.</p>

<p>In other words, some kind of weight for each slave which would determine how often a slave is used.</p>

<p>Thanks</p>

---
<details><summary><i>Originally reported by <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=ebann">ebann</a>, imported from: <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-8423" target="_blank">Slave priority</a></i></summary>
<i><ul>
<li><b>assignee</b>: <img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/kohsuke.png" title="kohsuke's avatar" /> <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=kohsuke">kohsuke</a>
<li><b>status</b>: Resolved
<li><b>priority</b>: Major
<li><b>component(s)</b>: core
<li><b>resolution</b>: Won't Fix
<li><b>resolved</b>: 2014-05-18T23:56:48+00:00
<li><b>votes</b>: 25
<li><b>watchers</b>: 23
<li><b>imported</b>: 2025-11-23
</ul></i>
<details><summary>Raw content of original issue</summary>

<pre>
<p>Hello,</p>

<p>I have several slaves to execute my jobs.
Some are real user workstation, other are pure dedicated servers.</p>

<p>It would be great if I could say to Hudson:
"Use this dedicated slaves as much as you can. And if they are busy use the not dedicated ones."</p>

<p>With this I would be able to handle heavy loads (with the extra help of user workstations).
But when everything is calm, I won't bother the real users with my jobs.</p>

<p>In other words, some kind of weight for each slave which would determine how often a slave is used.</p>

<p>Thanks</p></pre>
</details>
</details>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/whitey04.png" title="whitey04's avatar" /> whitey04:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-8423?focusedId=166210&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-166210">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>+1</p>
</pre>
</details>

</ul>
</details>

<p>+1</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i>gandalf:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-8423?focusedId=173851&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-173851">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Definitely a weighting would be great as my cluster is made up of slow and fast machines, and often jobs run on the slow machine when a faster one is available to use. Tying jobs to particular nodes or sub clusters is not really an option as i have hundreds of jobs all with equal priority.</p>
</pre>
</details>

</ul>
</details>

<p>Definitely a weighting would be great as my cluster is made up of slow and fast machines, and often jobs run on the slow machine when a faster one is available to use. Tying jobs to particular nodes or sub clusters is not really an option as i have hundreds of jobs all with equal priority.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/kohsuke.png" title="kohsuke's avatar" /> kohsuke:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-8423?focusedId=177444&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-177444">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>A plugin can do this by implementing <tt>LoadBalancer</tt> and setting it to <tt>Queue</tt>.</p>
</pre>
</details>

</ul>
</details>

<p>A plugin can do this by implementing <tt>LoadBalancer</tt> and setting it to <tt>Queue</tt>.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/alex_ouzounis.png" title="alex_ouzounis's avatar" /> alex_ouzounis:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-8423?focusedId=178072&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-178072">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>+1</p>
</pre>
</details>

</ul>
</details>

<p>+1</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i>gandalf:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-8423?focusedId=178076&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-178076">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>thanks for the heads up on the plugin info. if i get some spare time i'll have a look at writing a plugin, havent written one before.</p>

</pre>
</details>

</ul>
</details>

<p>thanks for the heads up on the plugin info. if i get some spare time i'll have a look at writing a plugin, havent written one before.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i>arashi_gr:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-8423?focusedId=180744&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-180744">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>I've been using jenkins for almost a year now, and currently have 12 slaves connected.
This is hands down one of the most important upgrades that would greatly affect day to day operations.
Oddly enough , my current cross platform build machines consist of a combination of fast-slow-slow, so waiting on a 2-core build when an 8-core machine is idling, kinda looks like squandered resources.</p>
</pre>
</details>

</ul>
</details>

<p>I've been using jenkins for almost a year now, and currently have 12 slaves connected.<br/>
This is hands down one of the most important upgrades that would greatly affect day to day operations.<br/>
Oddly enough , my current cross platform build machines consist of a combination of fast-slow-slow, so waiting on a 2-core build when an 8-core machine is idling, kinda looks like squandered resources.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/rx14.png" title="rx14's avatar" /> rx14:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-8423?focusedId=192168&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-192168">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>This needs to happen. I have a small linux server hosting my Jenkins which takes some time building and slows down everything else on the server while doing so and I also have my Desktop which is much beefier. I would like to have the jobs run on my PC when it's online but on the linux server when my PC is off...</p>
</pre>
</details>

</ul>
</details>

<p>This needs to happen. I have a small linux server hosting my Jenkins which takes some time building and slows down everything else on the server while doing so and I also have my Desktop which is much beefier. I would like to have the jobs run on my PC when it's online but on the linux server when my PC is off...</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i>gandalf:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-8423?focusedId=192170&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-192170">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Sadly I still haven't found the time to look into writing this.</p>
</pre>
</details>

</ul>
</details>

<p>Sadly I still haven't found the time to look into writing this.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i>dimmon:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-8423?focusedId=193721&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-193721">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>I have found recently developed plugin with this functionality:
Scoring Load Balancer plugin
<a href="https://wiki.jenkins-ci.org/display/JENKINS/Scoring+Load+Balancer+plugin" class="external-link" target="_blank" rel="nofollow noopener">https://wiki.jenkins-ci.org/display/JENKINS/Scoring+Load+Balancer+plugin</a></p>
</pre>
</details>

</ul>
</details>

<p>I have found recently developed plugin with this functionality:<br/>
Scoring Load Balancer plugin<br/>
<a href="https://wiki.jenkins-ci.org/display/JENKINS/Scoring+Load+Balancer+plugin" class="external-link" target="_blank" rel="nofollow noopener">https://wiki.jenkins-ci.org/display/JENKINS/Scoring+Load+Balancer+plugin</a></p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/danielbeck.png" title="danielbeck's avatar" /> danielbeck:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-8423?focusedId=200788&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-200788">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>As this can easily be resolved in plugins (and one is already mentioned in the comments), there is no need to resolve this in core. Therefore marking as Won't Fix.</p>
</pre>
</details>

</ul>
</details>

<p>As this can easily be resolved in plugins (and one is already mentioned in the comments), there is no need to resolve this in core. Therefore marking as Won't Fix.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i>markjanes:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-8423?focusedId=228855&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-228855">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>If you just have 2 categories of slaves (fast/slow), you can configure the slow slave's "Availability" setting to take the slave off-line when idle.</p>

<p>Then jobs should go to the faster slaves unless jobs are sitting in the queue.</p>
</pre>
</details>

</ul>
</details>

<p>If you just have 2 categories of slaves (fast/slow), you can configure the slow slave's "Availability" setting to take the slave off-line when idle.</p>

<p>Then jobs should go to the faster slaves unless jobs are sitting in the queue.</p>
